components/reddit/actions/list-comments/list-comments.mjs

```diff
@@ -0,0 +1,89 @@
+import reddit from "../../reddit.app.mjs";
+import get from "lodash/get.js";
+import { axios } from "@pipedream/platform";
+
+export default {
+  type: "action",
+  key: "reddit-action-list-comments",
+  version: "0.0.1",
+  name: "List Comments",
+  description: "List comments for a specific post. [See the docs here](https://www.reddit.com/dev/api/#GET_comments_{article})",
+  props: {
+    reddit,
+    subReddit: {
+      propDefinition: [
+        reddit,
+        "subreddit",
+      ],
+    },
+    subRedditPost: {
+      propDefinition: [
+        reddit,
+        "subredditPost",
+        ({ subReddit }) => ({
+          subReddit,
+        }),
+      ],
+    },
+    numberOfParents: {
+      propDefinition: [
+        reddit,
+        "numberOfParents",
+      ],
+    },
+    depth: {
+      propDefinition: [
+        reddit,
+        "depth",
+      ],
+    },
+    includeSubredditDetails: {
+      propDefinition: [
+        reddit,
+        "includeSubredditDetails",
+      ],
+    },
+    sort: {
+      type: "string",
+      label: "Sort",
+      description: "The order of the comments",
+      optional: true,
+      options: [
+        "confidence",
+        "top",
+        "new",
+        "controversial",
+        "old",
+        "random",
+        "qa",
+        "live",
+      ],
+    },
+  },
+  async run({ $ }) {
+    const {
+      subReddit,
+      subRedditPost,
+      numberOfParents,
+      depth,
+      includeSubredditDetails,
+      sort,
+    } = this;
+
+    const res = await axios($, this.reddit._getAxiosParams({
+      method: "GET",
+      path: `/r/${get(subReddit, "value", subReddit)}/comments/article`,
+      params: {
+        article: get(subRedditPost, "value", subRedditPost),
+        context: numberOfParents,
+        depth: depth,
+        sr_detail: includeSubredditDetails,
+        threaded: true,
+        sort: sort,
+      },
+    }));
```

Comment:
Could we paginate the comments in a post using the [`after` parameter](https://www.reddit.com/dev/api/#listings)? For example:
- Add a `max` prop like in [Airtable `common-list.js`](https://github.com/PipedreamHQ/pipedream/blob/master/components/airtable/actions/common-list.js#L18)
- Paginate comments by making API requests until there are no more comments or we've retrieved `max` comments
- Optionally add an `after` prop to set as the initial `after` parameter in the request

Reply:
I think it can be a little dangerous if the user puts a high number in the limit. Reddit has a very aggressive rate limit. Besides, I think we should leave the user to paginate it the way he wants it. In Reddit you can use a `before` or `after` logic. I added these two variables and the `limit` variable. Please let me know if you disagree with my new approach